.cirrus.yml

```diff
@@ -0,0 +1,180 @@
+env:
+  WIDEMUL: auto
+  BIGNUM: auto
+  STATICPRECOMPUTATION: yes
+  ECMULTGENPRECISION: auto
+  ASM: no
+  BUILD: check
+  WITH_VALGRIND: yes
+  RUN_VALGRIND: no
+  EXTRAFLAGS:
+  HOST:
+  ECDH: no
+  RECOVERY: no
+  SCHNORRSIG: no
+  EXPERIMENTAL: no
+  CTIMETEST: yes
+  BENCH: yes
+  ITERS: 2
+  # We only need the top commit
+  CIRRUS_CLONE_DEPTH: 1
+
+cat_logs_snippet: &CAT_LOGS
+  always:
+    test_logs_script:
+      - cat tests.log || true
+      - cat exhaustive_tests.log || true
+      - cat valgrind_ctime_test.log || true
+      - cat bench.log || true
+  on_failure:
+    debug_output_script:
+      - cat config.log || true
+      - cat test_env.log || true
+      - env
+
+docker_arg_force_build_snippet: &DOCKER_ARGS_FORCE_REBUILD
+  docker_arguments:
+  # Add build id to the key of the Docker cache in order to force rebuild of Docker image for every build.
```

Comment:
Yes, this deserves some explanation.

> Cirrus CI is relies on Docker images for Linux builds. We make use of this extensively by building our own Docker images (with necessary packages) on which CI then runs (https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment), see the dockerfile setting in .cirrus.yml. The image is recreated for every build and then all "tasks" run on this image.

Without this (somewhat hackish) line, the Docker image will be recreated only when the Dockerfile changes. 

I initially added this with the idea of building with a bleeding edge toolchains in mind. Now that the plan is to do these builds differently, clearing the cache is not as important. I still think it's a good idea because it adds only a few minutes to the build and there's no button to clear the caches (https://github.com/cirruslabs/cirrus-ci-docs/issues/544).

BUT I just noticed that this line doesn't work at all and the reason is that Cirrus simply reverted the change which made it work (https://github.com/cirruslabs/cirrus-cli/pull/221#issuecomment-756814062). So I'll remove it...

We'll need to change the Dockerfile then if we really want to force a rebuild, at least until the aforementioned Cirrus issue has been solved. Not elegant but acceptable.